src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts

```diff
@@ -804,6 +839,10 @@ export class ProgrammingExerciseUpdateComponent implements AfterViewInit, OnDest
         }
     }
 
+    protected switchEditMode = () => {
+        this.isSimpleMode.set(!this.isSimpleMode());
```

Comment:
just for info (completely optional): you could also use `this.isSimpleMode.update((isSimpleMode) => !isSimpleMode)`

Reply:
Thanks for the heads-up - following [this explanation](https://www.angulartraining.com/daily-newsletter/three-ways-to-update-angular-signals/) `update` makes more sense here